api/ocm/plugin/internal/transfer.go

```diff
@@ -0,0 +1,91 @@
+package internal
+
+import (
+	"encoding/json"
+
+	metav1 "ocm.software/ocm/api/ocm/compdesc/meta/v1"
+	v2 "ocm.software/ocm/api/ocm/compdesc/versions/v2"
+	ocm "ocm.software/ocm/api/ocm/cpi"
+	"ocm.software/ocm/api/tech"
+)
+
+const (
+	Q_UPDATE_VERSION    = "updateversion"
+	Q_OVERWRITE_VERSION = "overwriteversion"
+	Q_ENFORCE_TRANSPORT = "enforcetransport"
+	Q_TRANSFER_VERSION  = "transferversion"
+	Q_TRANSFER_RESOURCE = "transferresource"
+	Q_TRANSFER_SOURCE   = "transfersource"
+)
+
+type UniformAccessSpecInfo = tech.UniformAccessSpecInfo
+
+type SourceComponentVersion struct {
+	Name       string                    `json:"component"`
+	Version    string                    `json:"version"`
+	Provider   metav1.Provider           `json:"provider,omitempty"`
+	Repository ocm.GenericRepositorySpec `json:"repository"`
+	Labels     metav1.Labels             `json:"labels,omitempty"`
+}
+
+type TargetRepositorySpec = ocm.GenericRepositorySpec
+
+type TransferOptions struct {
+	Recursive         *bool            `json:"recursive,omitempty"`
+	ResourcesByValue  *bool            `json:"resourcesByValue,omitempty"`
+	LocalByValue      *bool            `json:"localByValue,omitempty"`
+	SourcesByValue    *bool            `json:"sourcesByValue,omitempty"`
+	KeepGlobalAccess  *bool            `json:"keepGlobalAccess,omitempty"`
+	StopOnExisting    *bool            `json:"stopOnExisting,omitempty"`
+	EnforceTransport  *bool            `json:"enforceTransport,omitempty"`
+	Overwrite         *bool            `json:"overwrite,omitempty"`
+	SkipUpdate        *bool            `json:"skipUpdate,omitempty"`
+	OmitAccessTypes   []string         `json:"omitAccessTypes,omitempty"`
+	OmitArtifactTypes []string         `json:"omitArtifactTypes,omitempty"`
+	Special           *json.RawMessage `json:"special,omitempty"`
+}
+
+// Resolution describes the transport context for a component
+// version, including the new handler specification and
+// the source repository to use to look up the reference.
+type Resolution struct {
+	RepositorySpec *ocm.GenericRepositorySpec `json:"repository,omitempty"`
+	// TransferHandler is the handler identity according to the transfer handler
+	// name scheme.
+	TransferHandler string           `json:"transferHandler,omitempty"`
+	TransferOptions *TransferOptions `json:"transferOptions,omitempty"`
```

Comment:
add documentation why the transfer handlers returns the transfer options (= because it might change the options).